app/code/GorbanSv/ProductCustomFields/Ui/Component/Form/DataProvider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace GorbanSv\ProductCustomFields\Ui\Component\Form;
+
+use Magento\Cron\Model\ResourceModel\Schedule\CollectionFactory;
+
+/**
+ * Class DataProvider
+ */
+class DataProvider extends \Magento\Ui\DataProvider\AbstractDataProvider
+{
+    /**
+     * @var \Magento\Cron\Model\ResourceModel\Schedule\Collection
+     */
+    protected $collection;
+    /**
+     * @var array
+     */
+    protected $loadedData;
+    /**
+     * Constructor
+     *
+     * @param string $name
+     * @param string $primaryFieldName
+     * @param string $requestFieldName
+     * @param CollectionFactory $blockCollectionFactory
+     * @param array $meta
+     * @param array $data
+     */
+    public function __construct(
+        $name,
+        $primaryFieldName,
+        $requestFieldName,
+        CollectionFactory $blockCollectionFactory,
+        array $meta = [],
+        array $data = []
+    ) {
+        $this->collection = $blockCollectionFactory->create();
+        parent::__construct($name, $primaryFieldName, $requestFieldName, $meta, $data);
+    }
+    /**
+     * Get data
+     *
+     * @return array
+     */
+    public function getData()
+    {
+        return [];
+    }
+    public function getMeta()
+    {
+        $meta = parent::getMeta();
+        $meta['general']['children']['custom_field'] = [
```

Comment:
While trying to edit the product in Admin I'm experiencing the error:
![error](http://dl4.joxi.net/drive/2019/01/29/0033/0390/2195846/46/a85697aef0.jpg)
Seems the metadata modification in your data provider GorbanSv/ProductCustomFields/Ui/Component/Form/DataProvider::getMeta is not correct:
![debug](http://dl3.joxi.net/drive/2019/01/29/0033/0390/2195846/46/c6ca36111d.jpg)
While trying the quick fix
https://gist.github.com/chabanenk0/3ccb8599c7ab95c942904c5c55517582
I was able to load the product edit page in Admin:
but there is no original fields:
![no product edit](http://dl4.joxi.net/drive/2019/01/29/0033/0390/2195846/46/0f14b57d32.jpg)
Please fix your data provider in order not to remove the original components of the product section in Magento Admin:
![proper admin](http://dl3.joxi.net/drive/2019/01/29/0033/0390/2195846/46/2781052e56.jpg)
Seems your ui configuration app/code/GorbanSv/ProductCustomFields/view/adminhtml/ui_component/product_form.xml
overwrites the original Magento one:
https://github.com/magento/magento2/blob/2.2/app/code/Magento/Catalog/view/adminhtml/ui_component/product_form.xml
Try to modify your file product_form.xml to avoid original data overlap. Not sure what is the best way to implement it correctly, try investigating and let me know the result.